sopel/run_script.py

```diff
@@ -148,19 +149,22 @@ def build_parser():
 
 
 def check_not_root():
+
     """Check if root is running the bot.
 
     It raises a ``RuntimeError`` if the user has root privileges on Linux or
     if it is the ``Administrator`` account on Windows.
     """
-    try:
+    if platform.system() in ["Linux", "Darwin"]:
         # Linux/Mac
         if os.getuid() == 0 or os.geteuid() == 0:
             raise RuntimeError('Error: Do not run Sopel with root privileges.')
-    except AttributeError:
+    elif platform.system() in ["Windows"]:
         # Windows
         if os.environ.get("USERNAME") == "Administrator":
             raise RuntimeError('Error: Do not run Sopel as Administrator.')
+    else:
+        raise RuntimeError('Error: Could not detect Operating Sytem Type, or it is unsupported.')
```

Comment:
I wouldn't raise an exception, but trigger a warning instead: an uncommon OS might run fine, and user may need to get a notification - that might raise questions, for sure, but also provide Sopel's dev a feedback on which OS is supported and how to support it (if it ever happens).

I think [RuntimeWarning](https://docs.python.org/3/library/exceptions.html#RuntimeWarning) would be a good fit.

See: https://docs.python.org/3.7/library/warnings.html?highlight=warning#available-functions

Reply:
:+1:, for example Solaris based systems should be able to run Sopel, but this error would preclude that.

Reply:
used stderr from sopel.tools